docs/developer-guidelines.md

```diff
@@ -0,0 +1,172 @@
+# Overview
+
+This document represents the developer policies and procedures for maintaining
+Hardware as a Service (HaaS), including the accepted [coding style](#Coding-style).
+
+Welcome! Overall, we follow the [github fork & pull
+model](http://scottchacon.com/2011/08/31/github-flow.html), where users fork
+the [main HaaS repository][repo], push changes to their personal fork and then
```

Comment:
Should we mention that the user should create a new branch other than using the master branch in their personal fork each time when the user trying to make changes?